Pod/Classes/PINAnimatedImageManager.m

```diff
@@ -282,55 +323,81 @@ + (void)processAnimatedImage:(NSData *)animatedImageData
           //write empty frame count
-          [fileHandle writeData:[NSData dataWithBytes:&frameCountForFile length:sizeof(frameCountForFile)]];
+          @try {
+            [fileHandle writeData:[NSData dataWithBytes:&frameCountForFile length:sizeof(frameCountForFile)]];
+          } @catch (NSException *exception) {
+            NSError *frameCountError = [NSError errorWithDomain:kPINAnimatedImageErrorDomain code:PINAnimatedImageErrorFileWrite userInfo:exception];
+            HANDLE_PROCESSING_ERROR(frameCountError);
+          } @finally {}
         });
         
         //Process frames
         for (NSUInteger frameIdx = 0; frameIdx < frameCount; frameIdx++) {
+          if (PROCESSING_ERROR != nil) {
+            break;
+          }
           @autoreleasepool {
             if (fileDuration > maxFileDuration || fileSize > maxFileSize) {
               //create a new file
               dispatch_group_async(diskGroup, diskWriteQueue, ^{
                 //prepend file with frameCount
-                [fileHandle seekToFileOffset:0];
-                [fileHandle writeData:[NSData dataWithBytes:&frameCountForFile length:sizeof(frameCountForFile)]];
-                [fileHandle closeFile];
+                @try {
+                  [fileHandle seekToFileOffset:0];
+                  [fileHandle writeData:[NSData dataWithBytes:&frameCountForFile length:sizeof(frameCountForFile)]];
+                  [fileHandle closeFile];
+                } @catch (NSException *exception) {
+                  NSError *frameCountError = [NSError errorWithDomain:kPINAnimatedImageErrorDomain code:PINAnimatedImageErrorFileWrite userInfo:exception];
+                  HANDLE_PROCESSING_ERROR(frameCountError);
+                } @finally {}
               });
               
               dispatch_group_async(diskGroup, diskWriteQueue, ^{
                 PINLog(@"notifying file: %@", filePath);
-                completion(NO, filePath, error);
+                completion(NO, filePath, PROCESSING_ERROR);
               });
               
               diskGroup = dispatch_group_create();
               fileCount++;
-              fileHandle = [self fileHandle:&error filePath:&filePath temporaryDirectory:temporaryDirectory UUID:UUID count:fileCount];
+              NSError *fileHandleError = nil;
+              fileHandle = [self fileHandle:&fileHandleError filePath:&filePath temporaryDirectory:temporaryDirectory UUID:UUID count:fileCount];
+              HANDLE_PROCESSING_ERROR(fileHandleError);
               frameCountForFile = 0;
               fileDuration = 0;
               fileSize = 0;
               //write empty frame count
               dispatch_group_async(diskGroup, diskWriteQueue, ^{
-                [fileHandle writeData:[NSData dataWithBytes:&frameCountForFile length:sizeof(frameCountForFile)]];
+                @try {
+                  [fileHandle writeData:[NSData dataWithBytes:&frameCountForFile length:sizeof(frameCountForFile)]];
```

Comment:
Are you asking if it should be closed here? It's actually just been opened and we're writing empty bytes just to advance the internal pointer. Before we close the file, we backtrack to the beginning and write the correct frame count.